tests/cluster/skips.go

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package cluster
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/client-go/kubecli"
+)
+
+func SkipWhenClusterHasASingleNode(virtClient kubecli.KubevirtClient) {
+	clusterNodes := GetAllSchedulableNodes(virtClient)
```

Comment:
@EdDev for migration I don't think there's a need to compare the cached list of nodes w/ the current number of schedulable nodes - migration fails when you have no node to move to - and the error msg is quite obvious.

I am forced to agree that this is a bit more generic than that, and as such, I'd probably just be better off renaming this (and update the "skip" msg) to something migration related.

Your thoughts ?

Reply:
I am not really sure what is your strategy here. If the check here is about the number of schedulable nodes, then this is the best name we can think of. Hiding the this under a different name will do nothing except abstracting, and I am not sure this one merits abstraction.

The error of a failed migration is not helpful, as this attempts to avoid that from ever happening. It can be relevant if you do something similar to the XFail solution, but with a check on the error. In that case, you can skip in case a specific error occurs.

And finally about the cache, it was about identifying if we skip for the right reason. The setup may have started in a good shape, then a node would be in maintenance or recovering from some state and we would unintentionally skip the test. My suggestion was that on setup, we detect the number of schedule-able nodes and based on it take a decisions. I must admit, that even if you do test on setup, it feels like we may be in a race here (e.g. not all nodes are ready).

There is another option: You label the migration tests, then the script that starts the tests may exclude them if there is only one node.